src/app/public/src/modules/shared/route.service.ts

```diff
@@ -126,36 +126,45 @@ export class StacheRouteService {
   }
 
   private sortRoutes(routes: StacheNavLink[]): StacheNavLink[] {
-
-    const sortByName = (a: any, b: any): number  => {
-      if (a.name.toLowerCase() < b.name.toLowerCase()) {
-        return -1;
-      } else if (a.name.toLowerCase() > b.name.toLowerCase()) {
-        return 1;
-      } else {
-        return 0;
-      }
-    };
-
-    const sortByOrder = (a: any, b: any): number => {
-      if (a.order < b.order) {
-        return -1;
-      } else if (a.order > b.order) {
-        return 1;
+    const sortedRoutes = routes.filter(route => !route.hasOwnProperty('order'))
+      .sort(this.sortByName);
+
+    const routesWithNavOrder = routes.filter(route => route.hasOwnProperty('order'))
+      .sort(this.sortByName)
+      .sort(this.sortByOrder);
+
+    routesWithNavOrder.forEach(route => {
+      let newIdx = route.order - 1;
+      if (newIdx < sortedRoutes.length) {
+        while (newIdx < sortedRoutes.length && sortedRoutes[newIdx].order <= route.order) {
```

Comment:
I'm sure this logic is sound, but I had a difficult time dissecting it. If there's any way to simplify with inline methods or wrapping expressions in a single boolean variable, it would help my aging brain. Love the demo you created for this! Helps visualize exactly what you're going for.